scripts/scan_memgraph.py

```
@@ -0,0 +1,246 @@
+
+    # Some targets produce no JSON at all; check before opening.
+    if not os.path.isfile(output_path):
+        raise RuntimeError(
+            f"Expected JSON not found for {target!r}; no file at {output_path}"
+        )
+
+    try:
+        with open(output_path, "r", encoding="utf-8") as f:
+            data = json.load(f)
+    except (OSError, json.JSONDecodeError) as e:
+        raise RuntimeError(f"Could not read JSON for {target!r}: {e}")
```

Ruff (0.11.9)

Within an `except` clause, raise exceptions with `raise ... from err` or `raise ... from None` to distinguish them from errors in exception handling

(B904)